idea/testData/intentions/convertTwoComparisonsToRangeCheck/compareToDifferentTypes.kt

```diff
@@ -0,0 +1,9 @@
+// IS_APPLICABLE: false
+// WITH_RUNTIME
+class A(val _value: Int) {
+    operator fun compareTo(other: Int) = _value.compareTo(other)
+}
+
+fun test(a: A): Boolean {
+    return a >= 0 && a <= 100<caret>
```

Comment:
The class defines compareTo(Int), so the code is green, but intention makes it red. Is it too esoteric to detect?